indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,7 +155,9 @@ public IndexTask(
       @JsonProperty("resource") TaskResource taskResource,
       @JsonProperty("spec") IndexIngestionSpec ingestionSchema,
       @JacksonInject ObjectMapper jsonMapper,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JacksonInject IndexMerger indexMerger,
+      @JacksonInject IndexIO indexIO
```

Comment:
Fwiw I think tranquility will probably be able to handle these, as long as its injector has the ability to make these classes (which if it doesn't, could be added). Putting them in the TaskToolbox is cool too if we want to go that way.